src/query/tsdb/remote/server.go

```diff
@@ -72,35 +77,32 @@ func (s *grpcServer) Fetch(
 	message *rpc.FetchRequest,
 	stream rpc.Query_FetchServer,
 ) error {
-	ctx := RetrieveMetadata(stream.Context())
+	ctx := retrieveMetadata(stream.Context())
 	logger := logging.WithContext(ctx)
-	storeQuery, err := DecodeFetchRequest(message)
+	storeQuery, err := decodeFetchRequest(message)
 	if err != nil {
-		logger.Error("unable to decode fetch query", zap.Any("error", err))
+		logger.Error("unable to decode fetch query", zap.Error(err))
 		return err
 	}
 
-	result, cleanup, err := s.storage.FetchRaw(ctx, storeQuery, nil)
+	result, cleanup, err := s.storage.FetchCompressed(
+		ctx,
+		storeQuery,
+		storage.NewFetchOptions(),
+	)
 	defer cleanup()
 	if err != nil {
 		logger.Error("unable to fetch local query", zap.Error(err))
 		return err
 	}
 
-	available, pools, err, poolCh, errCh := s.poolWrapper.IteratorPools()
+	pools, err := s.poolWrapper.WaitForIteratorPools(poolTimeout)
```

Comment:
instead of using a separate pooltimeout, maybe use the ctx timeout ?

Reply:
I'd rather keep this separate; it should never really have to wait since the iterator pools should spin up very soon after startup